flask_resources/errors.py

```diff
@@ -75,47 +117,64 @@ def get_body(self, environ=None):
         return body
 
 
-# FIXME: From here down need review
-
+class SchemaValidationError(RESTException):
+    """Marshmallow schema validation exception."""
 
-class FieldError(object):
-    """Represents a field level error.
+    def __init__(self, errors=None, **kwargs):
+        """Initialize exception."""
+        _errors = []
+        if errors:
+            for field, messages in errors.items():
+                _errors.extend([_FieldError(field, msg) for msg in messages])
+        super(SchemaValidationError, self).__init__(errors=_errors, **kwargs)
 
-    .. note:: This is not an actual exception.
-    """
 
-    def __init__(self, field, message, code=None):
-        """Init object.
+    class _FieldError(object):
```

Comment:
> Since data validation is a concept introduced in `invenio-records-resource` shouldn't this exception be defined there? Not in the service errors, but in the global (resources) errors.

Same applies in this case

Reply:
It's rather that it shouldn't be called Schema error.... it's a generic REST API error, and defines an interface for how to expose certain type of more specific errorrs